### ISSUE TITLE
Payara-Embedded: Servlet [RegistrationCoordinatorPortTypePortImpl] and Servlet [RegistrationRequesterPortTypePortImpl] have the same url pattern

### ISSUE THREAD

**Issue by UlrichCech** [payara/Payara#1120]
Even with the smallest possible REST-Application (one GET-Endpoint), the starting of the embedded Payara (4.1.1.163.0.1) fails with the following stacktrace.
I tested with 4.1.1.162/4.1.1.163 as well.

If this small WAR is deployed to a Payara Full Server 162/163, everything works fine.

The test-project can be found here:
https://github.com/UlrichCech/payara-embedded-error

```
SCHWERWIEGEND: Servlet [RegistrationCoordinatorPortTypePortImpl] and Servlet [RegistrationRequesterPortTypePortImpl] have the same url pattern: [/RegistrationService_V10]
Okt 04, 2016 11:10:44 AM org.glassfish.api.ActionReport failure
SCHWERWIEGEND: Exception while deploying the app [payara-embedded-1.0-SNAPSHOT]
Okt 04, 2016 11:10:44 AM com.sun.enterprise.v3.server.ApplicationLifecycle deploy
SCHWERWIEGEND: Exception during lifecycle processing
java.lang.IllegalStateException: Servlet [RegistrationCoordinatorPortTypePortImpl] and Servlet [RegistrationRequesterPortTypePortImpl] have the same url pattern: [/RegistrationService_V10]. Related annotation information: annotation [@javax.jws.WebService(name=, wsdlLocation=/wsdls/wsc10/wscoor.wsdl, portName=RegistrationRequesterPortTypePort, endpointInterface=com.sun.xml.ws.tx.coord.v10.types.RegistrationRequesterPortType, serviceName=RegistrationService_V10, targetNamespace=http://schemas.xmlsoap.org/ws/2004/10/wscoor)] on annotated element [class com.sun.xml.ws.tx.coord.v10.endpoint.RegistrationRequesterPortTypePortImpl] of type [TYPE]
	at com.sun.enterprise.deployment.archivist.Archivist.readAnnotations(Archivist.java:519)
	at com.sun.enterprise.deployment.archivist.Archivist.readAnnotations(Archivist.java:447)
	at org.glassfish.web.deployment.archivist.WebArchivist.postAnnotationProcess(WebArchivist.java:339)
	at org.glassfish.web.deployment.archivist.WebArchivist.postAnnotationProcess(WebArchivist.java:92)
	at com.sun.enterprise.deployment.archivist.Archivist.readRestDeploymentDescriptors(Archivist.java:421)
	at com.sun.enterprise.deployment.archivist.Archivist.openWith(Archivist.java:292)
	at com.sun.enterprise.deployment.archivist.ApplicationFactory.openWith(ApplicationFactory.java:232)
	at org.glassfish.javaee.core.deployment.DolProvider.processDOL(DolProvider.java:193)
	at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:227)
	at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:96)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.loadDeployer(ApplicationLifecycle.java:882)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:822)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:378)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:220)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:487)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:539)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:535)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:534)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:557)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:556)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1464)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:109)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1846)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1722)
	at com.sun.enterprise.admin.cli.embeddable.DeployerImpl.deploy(DeployerImpl.java:134)
	at de.sofiworx.BootstrapServer.main(BootstrapServer.java:21)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:294)
	at java.lang.Thread.run(Thread.java:745)
Caused by: Servlet [RegistrationCoordinatorPortTypePortImpl] and Servlet [RegistrationRequesterPortTypePortImpl] have the same url pattern: [/RegistrationService_V10]. Related annotation information: annotation [@javax.jws.WebService(name=, wsdlLocation=/wsdls/wsc10/wscoor.wsdl, portName=RegistrationRequesterPortTypePort, endpointInterface=com.sun.xml.ws.tx.coord.v10.types.RegistrationRequesterPortType, serviceName=RegistrationService_V10, targetNamespace=http://schemas.xmlsoap.org/ws/2004/10/wscoor)] on annotated element [class com.sun.xml.ws.tx.coord.v10.endpoint.RegistrationRequesterPortTypePortImpl] of type [TYPE]
	at org.glassfish.apf.impl.AnnotationProcessorImpl.process(AnnotationProcessorImpl.java:375)
	at org.glassfish.apf.impl.AnnotationProcessorImpl.process(AnnotationProcessorImpl.java:383)
	at org.glassfish.apf.impl.AnnotationProcessorImpl.processAnnotations(AnnotationProcessorImpl.java:291)
	at org.glassfish.apf.impl.AnnotationProcessorImpl.process(AnnotationProcessorImpl.java:196)
	at org.glassfish.apf.impl.AnnotationProcessorImpl.process(AnnotationProcessorImpl.java:135)
	at com.sun.enterprise.deployment.archivist.Archivist.processAnnotations(Archivist.java:627)
	at org.glassfish.web.embed.impl.EmbeddedWebArchivist.processAnnotations(EmbeddedWebArchivist.java:149)
	at com.sun.enterprise.deployment.archivist.Archivist.readAnnotations(Archivist.java:463)
	... 36 more
Caused by: java.lang.IllegalArgumentException: Servlet [RegistrationCoordinatorPortTypePortImpl] and Servlet [RegistrationRequesterPortTypePortImpl] have the same url pattern: [/RegistrationService_V10]
	at org.glassfish.web.deployment.descriptor.WebBundleDescriptorImpl.addWebComponentDescriptor(WebBundleDescriptorImpl.java:370)
	at org.glassfish.webservices.connector.annotation.handlers.WebServiceHandler.processAnnotation(WebServiceHandler.java:461)
	at com.sun.enterprise.deployment.annotation.factory.SJSASFactory$LazyAnnotationHandler.processAnnotation(SJSASFactory.java:148)
	at org.glassfish.apf.impl.AnnotationProcessorImpl.process(AnnotationProcessorImpl.java:352)
	... 43 more
```

**Comment by mikecroft** [comment]
Your ApplicationConfig doesn't implement `getClasses()`. Could you try adding that method [like in this example](http://tomee.apache.org/examples-trunk/rest-example-with-application/README.html) and share your results?

```Java
    public Set<Class<?>> getClasses() {
        return new HashSet<Class<?>>(Arrays.asList(de.sofiworx.users.boundary.UsersResource.class));
    }
```

**Comment by UlrichCech** [comment]
I added the getClasses() method, but nothing changed...

**Comment by smillidge** [comment]
The war you are generating contains payara embedded within it. Payara embedded should have provided scope when creating the war file.

**Comment by UlrichCech** [comment]
Hmm, but that's the main goal of the embedded version, that the Payara is IN the WAR-file and can be startet via a main-class, as provided in this blog post:
http://blog.payara.fish/using-payara-embedded-and-payara-micro-with-maven

And if I set the scope to "provided", if I start the main-class, I get "java.lang.NoClassDefFoundError: org/glassfish/embeddable/GlassFishException"... and that's the same problem as the user in the last comment to the mentioned blog-post describes.

**Comment by smillidge** [comment]
You will still need to add the Payara embedded jar onto the classpath of your application with the main class. Payara embedded shouldn't be included in your war file. If you are looking to create an executable uber jar I would recommend using Payara Micro and take a look at these reseources;
http://blog.payara.fish/creating-uber-jar-with-payara-micro
https://github.com/payara/Payara-Examples/tree/master/Payara-Micro/deployment-example
https://payara.gitbooks.io/payara-server/content/documentation/payara-micro/payara-micro.html

**Comment by UlrichCech** [comment]
Yes, with Payara Micro, I get it working, but then I don't need the Bootstrap-Server class anymore, but use command-line parameters for deploying and so on.

But now, I added an additional classpath element to the maven-exec-plugin (see changed GitHub-Test-project) and now it works.

But with this "hassle", I did not see any advantage for Payara Embedded and starting the server programmatically. Even for Docker-Environment, Payara-Micro seems much more useful.
